src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -77,9 +77,10 @@
      */
     /*package*/ final String encodedAuthorization;
 
-    private final Map<String,GHUser> users = new Hashtable<String, GHUser>();
-    private final Map<String,GHOrganization> orgs = new Hashtable<String, GHOrganization>();
-
+    private final Map<String,GHUser> users;
+    private final Map<String,GHOrganization> orgs;
+    // Cache of myself object.
+    private GHMyself myself;
```

Comment:
We now cache GHMySelf differently than the other users.  This is vs. keeping it in the user's map.  The issue is that if the first call to the user's map was for 'foouser' as a GHUser, then subsequent calls to getMsyelf() (if logged in as foouser) would always call the REST API because the type returned from the map would not be the desired type.  While theoretically we could just update the map with the more specific GHMyself (since it's a subclass of GHUser), I thought it better for clarity to simply cache it separately.